packages/core/rpc/ElectronRpcDriver.ts

```diff
@@ -62,80 +87,61 @@ class WindowWorkerHandle {
-        'Cannot use ElectronRpcDriver without electron available globally',
-      )
-    const electronRemote = electron.remote
+
+    this.config = config
+    this.channel = workerCreationChannel
+  }
+
+  makeWorker(): WindowWorkerHandle {
     const { ipcRenderer } = electronBetterIpc
-    if (!ipcRenderer)
+    if (!ipcRenderer) {
       throw new Error(
         'Cannot use ElectronRpcDriver without ipcRenderer from electron-better-ipc-extra',
       )
-
-    this.makeWorker = (): WindowWorkerHandle => {
-      const workerId = ipcRenderer.sendSync(workerCreationChannel)
-      const window = electronRemote.BrowserWindow.fromId(workerId)
-      const worker = new WindowWorkerHandle(ipcRenderer, window)
-      // out of band, take care of booting the worker
-      this.bootWorker(ipcRenderer, window, workerBootConfiguration).catch(
-        error => {
-          console.error('worker failed to boot')
-          console.error(error)
-        },
-      )
-      return worker
     }
-  }
-
-  // waits for a new worker to start, and then sends it its bootstrap configuration
-  async bootWorker(
-    ipcRenderer: NonNullable<typeof electronBetterIpc.ipcRenderer>,
-    window: Electron.BrowserWindow,
-    workerBootConfiguration: { plugins: PluginDefinition[] },
-  ) {
-    let readyForConfig = false
-    while (!readyForConfig) {
-      await wait(1000)
-      readyForConfig = !!(await ipcRenderer.callRenderer(
-        window,
-        'ready_for_configuration',
-      ))
+    if (!electron) {
+      throw new Error(
+        'Cannot use ElectronRpcDriver without electron available globally',
+      )
     }
-
-    const result = await ipcRenderer.callRenderer(
-      window,
-      'configure',
-      workerBootConfiguration,
-    )
-    if (!result) throw new Error('failed to configure worker')
+    const electronRemote = electron.remote
+    const workerId = ipcRenderer.sendSync(this.channel)
+    const window = electronRemote.BrowserWindow.fromId(workerId)
+    const worker = new WindowWorkerHandle(ipcRenderer, window, this.config)
+    return worker
   }
 
-  call(
+  async call(
     pluginManager: PluginManager,
     sessionId: string,
     functionName: string,
     args: {},
     options = {},
   ): Promise<unknown> {
-    return super
-      .call(pluginManager, sessionId, functionName, args, options)
-      .then(r => {
-        if (typeof r === 'object' && r !== null && 'imageData' in r) {
-          const img = new Image()
-          // @ts-ignore
-          img.src = r.imageData.dataURL
-          // @ts-ignore
-          r.imageData = img
-        }
-        return r
-      })
+    const r = await super.call(
+      pluginManager,
+      sessionId,
+      functionName,
+      args,
+      options,
+    )
+
+    if (typeof r === 'object' && r !== null && 'imageData' in r) {
+      const img = new Image()
+      // @ts-ignore
+      img.src = r.imageData.dataURL
+      // @ts-ignore
+      r.imageData = img
+    }
+    return r
```

Comment:
Wish we could clean this existing code up. It's really better for it to be tacked onto the end of `deserializeReturn` instead of `call`. Could you make that change while you're in here?

Reply:
I am not sure I know the path to doing that, the deserializeReturn applies to RpcMethodType, such as our renderers, but the electron app does not use custom renderers?

Reply:
oh you're right. the rpc driver doesn't have its own return deserialization. meh not worth worrying about right now, then.

Reply:
maybe the drivers need their own deserializeReturn